test-tools/proof-explorer/src/Root.tsx

```diff
@@ -0,0 +1,1006 @@
+                                            properties: {
+                                                attributes: {
+                                                    properties: object;
+                                                };
+                                            };
+                                        };
+                                    };
+                                };
+                                const attributes = schema.properties.credentialSubject.properties.attributes.properties;
+                                Object.entries(attributes).map(([tag, v]) => {
+                                    setTags((oldTags) => {
+                                        if (
+                                            oldTags.find(({ label }) => {
+                                                return label == tag;
+                                            })
+                                        ) {
+                                            return oldTags;
+                                        } else {
+                                            return [{ value: (v as { title: string }).title, label: tag }, ...oldTags];
+                                        }
+                                    });
+                                });
+                            } else {
+                                console.log(`Unable to get schema from ${schema_url}`);
+                            }
+                    }
+                } catch (e) {
+                    console.log(e);
+                }
+            }
+        };
+
+        fetchContracts().catch((err) => console.log(err));
+    }, [indexes]);
+
+    return [
+        tags,
+        <ul>
+            {issuers.map((idx) => (
+                <li> &lt;{idx.toString()},0&gt; </li>
+            ))}
+        </ul>,
+    ];
+}
+
+function parseIssuers(s: string): bigint[] {
+    return s
+        .split(',')
+        .filter((x) => x.trim().length != 0)
+        .map((x) => {
+            try {
+                return BigInt(parseInt(x));
+            } catch (e) {
+                return undefined;
+            }
+        })
+        .filter((x) => x != undefined)
+        .map((x) => x as bigint);
+}
+
+/**
+ * The main component.
+ */
+export default function ProofExplorer() {
+    const [provider, setProvider] = useState<WalletProvider>();
+
+    useEffect(() => {
+        if (provider !== undefined) {
+            return () => {
+                provider?.disconnect?.().then(() => provider.removeAllListeners());
+            };
+        }
+    }, [provider]);
+
+    const connectProvider = async (provider: WalletProvider) => {
+        await provider.connect();
+        setProvider(provider);
+    };
+
+    const [statement, setStatement] = useState<TopLevelStatements>([]);
+
+    const client = useRef(
+        new ConcordiumGRPCClient(new GrpcWebFetchTransport({ baseUrl: 'https://grpc.testnet.concordium.com:20000' }))
```

Comment:
The `baseURL` should be in a constant file or an env.

Reply:
Do you think there is value in that? Because this tool is meant for testnet, so it seems simplest to just hard-code these values.

What are you thinking would be the value in having it parametrized?

Reply:
I have now added it to the constants file.